### PR TITLE
Fix for RGW admin tool  with motr (user info)

### PR DESCRIPTION
**Pull request by sanalpk**
## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/common/options/rgw.yaml.in [hunk 1]
```diff
@@ -3261,6 +3261,14 @@ options:
   default: 0x7200000000000001:0x0
   services:
   - rgw
+- name: admin_motr_fid
+  type: str
+  level: advanced
+  desc: experimental Option to set my Motr fid
```
Comment (sumedhak27) [inline src/common/options/rgw.yaml.in:3267]:
```suggestion
  desc: Motr FID for admin-level access.
```

File: src/rgw/rgw_admin.cc [hunk 2]
```diff
@@ -4246,14 +4246,32 @@ int main(int argc, const char **argv)
 			   raw_period_update || raw_period_pull);
     bool need_cache = readonly_ops_list.find(opt_cmd) == readonly_ops_list.end();
     bool need_gc = (gc_ops_list.find(opt_cmd) != gc_ops_list.end()) && !bypass_gc;
+std::string rgw_store = "rados";
```
Comment (sumedhak27) [inline src/rgw/rgw_admin.cc:4249]:
```suggestion
    std::string rgw_store = "rados";
```

File: src/rgw/rgw_sal_motr.cc [hunk 3]
```diff
@@ -3441,15 +3441,30 @@ void *newMotrStore(CephContext *cct)
...
-    ldout(cct, 0) << "INFO: motr ha endpoint: " << ha_ep << dendl;
-    ldout(cct, 0) << "INFO: motr my fid:      " << proc_fid << dendl;
-    ldout(cct, 0) << "INFO: motr profile fid: " << profile << dendl;
+    const int init_flags = cct->get_init_flags();
+    ldout(cct, 0) << "INFO: init flags:        " << init_flags << dendl;
+    ldout(cct, 0) << "INFO: motr my endpoint:  " << proc_ep << dendl;
+    ldout(cct, 0) << "INFO: ha agent endpoint: " << ha_ep << dendl;
+    ldout(cct, 0) << "INFO: motr my fid:       " << proc_fid << dendl;
+    ldout(cct, 0) << "INFO: motr admin fid:    " << admin_proc_fid << dendl;
+    ldout(cct, 0) << "INFO: motr profile fid:  " << profile << dendl;
```
Comment (sumedhak27) [inline src/rgw/rgw_sal_motr.cc:3452]: we can increase the log level maybe before the final merge.

File: src/common/options/rgw.yaml.in [hunk 4]
```diff
@@ -3261,6 +3261,14 @@ options:
   default: 0x7200000000000001:0x0
   services:
   - rgw
+- name: admin_motr_fid
```
Comment (jjxsg) [inline src/common/options/rgw.yaml.in:3264]: We should have an entry for `admin_motr_endpoint` as well in `rgw.yaml.in`

File: src/common/options/rgw.yaml.in [hunk 5]
```diff
@@ -3261,6 +3261,14 @@ options:
...
+  type: str
+  level: advanced
+  desc: experimental Option to set my Motr fid
+  long_desc: example value 0x7200000000000001:0x29
+  default: 0x7200000000000001:0x0
+  services:
+  - rgw    
 - name: motr_my_endpoint
```
Comment (jjxsg) [inline src/common/options/rgw.yaml.in:3272]: Please add below entry for radosgw-admin motr endpoint
```
- name: admin_motr_endpoint
  type: str
  level: advanced
  desc: experimental Option to set my Motr endpoint
  long_desc: example value inet:tcp:10.230.241.24@5002
  default: inet:tcp:10.230.241.24@5002
  services:
  - rgw 
```

File: src/rgw/rgw_sal_motr.cc [hunk 6]
```diff
@@ -3441,15 +3441,30 @@ void *newMotrStore(CephContext *cct)
...
     store->conf.mc_local_addr  = proc_ep.c_str();
     store->conf.mc_process_fid = proc_fid.c_str();
+    // HACK this is so that radosge-admin uses a different client
+    if (init_flags == 0) {
+      store->conf.mc_process_fid = admin_proc_fid.c_str();
+    } else {
+      store->conf.mc_process_fid = proc_fid.c_str();
+    }
```
Comment (jjxsg) [inline src/rgw/rgw_sal_motr.cc:3460]: Please add admin endpoint in if condition
```
 if (init_flags == 0){
        store->conf.mc_process_fid = admin_proc_fid.c_str();
        store->conf.mc_local_addr = admin_proc_ep.c_str();
}
 else{
        store->conf.mc_process_fid = proc_fid.c_str();
        store->conf.mc_local_addr = proc_ep.c_str();
}
```

File: src/rgw/rgw_sal_motr.cc [hunk 7]
```diff
@@ -3441,15 +3441,30 @@ void *newMotrStore(CephContext *cct)
     const auto& proc_ep  = g_conf().get_val<std::string>("motr_my_endpoint");
     const auto& ha_ep    = g_conf().get_val<std::string>("motr_ha_endpoint");
     const auto& proc_fid = g_conf().get_val<std::string>("motr_my_fid");
+    const auto& admin_proc_fid = g_conf().get_val<std::string>("admin_motr_fid");
```
Comment (jjxsg) [inline src/rgw/rgw_sal_motr.cc:3444]: Please add
```
const auto &admin_proc_ep = g_conf().get_val<std::string>("admin_motr_endpoint");
```

**Review by jjxsg — CHANGES_REQUESTED** [review]

**Comment by sanalpk** [comment]
This changes are done as part of another ticket https://github.com/Seagate/cortx-rgw/pull/75/files

**Pull request closed without merging**